filter/producter.go

```diff
@@ -0,0 +1,28 @@
+package filter
+
+// Produter multiplies n numbers.
+type Produter struct {
```

Comment:
This is missing a `c`: `Producter`

Reply:
Also: what products are supplied on the output: the product of the last `n`, or a set of `n`?